### PR TITLE
Better usage of ccache

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Some translation units were always rebuilt regardless of ccache. The culprit is found and fixed.

### CODE REVIEW

File: src/Storages/System/StorageSystemBuildOptions.cpp.in [hunk 1]
```diff
@@ -8,7 +8,6 @@ const char * auto_config_build[]
     "SYSTEM", "@CMAKE_SYSTEM_NAME@",
     "VERSION_GITHASH", "@VERSION_GITHASH@",
     "VERSION_REVISION", "@VERSION_REVISION@",
-    "VERSION_DATE", "@VERSION_DATE@",
```
Comment (alexey-milovidov) [inline src/Storages/System/StorageSystemBuildOptions.cpp.in:11]: This field was not used (obsolete).

File: src/Common/config_version.h.in [hunk 2]
```diff
@@ -15,27 +14,10 @@
 #cmakedefine VERSION_OFFICIAL "@VERSION_OFFICIAL@"
 #cmakedefine VERSION_FULL "@VERSION_FULL@"
 #cmakedefine VERSION_DESCRIBE "@VERSION_DESCRIBE@"
-#cmakedefine VERSION_GITHASH "@VERSION_GITHASH@"
 #cmakedefine VERSION_INTEGER @VERSION_INTEGER@
-#cmakedefine VERSION_DATE @VERSION_DATE@
 
-#if defined(VERSION_MAJOR)
-#define DBMS_VERSION_MAJOR VERSION_MAJOR
```
Comment (alexey-milovidov) [inline src/Common/config_version.h.in:23]: This was redundant.

File: src/Common/config_version.h.in [hunk 3]
```diff
@@ -15,27 +14,10 @@
...
-#define DBMS_VERSION_MINOR 0
-#endif
-
-#if defined(VERSION_PATCH)
-#define DBMS_VERSION_PATCH VERSION_PATCH
-#else
-#define DBMS_VERSION_PATCH 0
-#endif
+/// These fields are frequently changing and we don't want to have them in the header file to allow caching.
+extern const char * VERSION_GITHASH;
```
Comment (alexey-milovidov) [inline src/Common/config_version.h.in:20]: The culprit is here.

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit fa9abc5038957ab2af34dc933825473231164ff4 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53184/fa9abc5038957ab2af34dc933825473231164ff4/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by davenger — APPROVED** [review]

**Pull request merged**